spec/virtual_total_spec.rb

```diff
@@ -328,4 +345,45 @@ def model_with_children(count)
       end
     end
   end
+
+  describe ".virtual_aggregation with has_many" do
+    let(:author) do
+      Author.create_with_books(1).tap do |author|
+        author.create_books(1, :published => true, :rating => 4)
+        author.create_books(1, :published => true, :rating => 2)
+        author.create_books(1, :published => true)
+      end
+    end
+
+    it "calculates sum locally" do
+      author
+      expect do
+        expect(author.sum_recently_published_books_rating).to eq(6)
+      end.to match_query_limit_of(1)
```

Comment:
Can we have the test data for this test make this a `match_query_limit_of(2)` instead of `1`.

The  "calculates sum in database" has less of an impact when this spec calls the same number of queries.

Reply:
@kbrock do you mean having 2 different authors? - so it shows the N+1?